runtime/compiler/aarch64/codegen/J9TreeEvaluator.cpp

```diff
@@ -101,6 +101,65 @@ J9::ARM64::TreeEvaluator::generateFillInDataBlockSequenceForUnresolvedField(TR::
    TR_ASSERT_FATAL(false, "This helper implements platform specific code for Fieldwatch, which is currently not supported on ARM64 platforms.\n");
    }
 
+TR::Register *
+J9::ARM64::TreeEvaluator::conditionalHelperEvaluator(TR::Node *node, TR::CodeGenerator *cg)
+   {
+
+   TR::RegisterDependencyConditions *conditions;
+   TR::Register *jumpReg, *valReg;
+   TR::Node *testNode = node->getFirstChild();
+   TR::Node *callNode = node->getSecondChild();
+   TR::Node *firstChild = testNode->getFirstChild();
+   TR::Node *secondChild = testNode->getSecondChild();
+   int32_t value, i, numArgs = callNode->getNumChildren();
+
+   conditions = new (cg->trHeapMemory()) TR::RegisterDependencyConditions(3, 3, cg->trMemory());
+
+   TR_ASSERT(numArgs <= 2, "Unexpected number of arguments for helper.");
+
+   // Helper arguments are in reversed order of the private linkage
+   // Argument registers are not needed to be split since the helper will
+   // preserve all of them.
+   int32_t iArgIndex = 0, fArgIndex = 0;
+   TR::Linkage *linkage = cg->createLinkage(TR_Private);
+   for (i = numArgs - 1; i >= 0; i--)
+      {
+      TR::Register *argReg = cg->evaluate(callNode->getChild(i));
+      TR::addDependency(conditions, argReg, (argReg->getKind() == TR_GPR) ? // Didn't consider Long here
+            linkage->getProperties().getIntegerArgumentRegister(iArgIndex++) : linkage->getProperties().getFloatArgumentRegister(fArgIndex++), argReg->getKind(), cg);
+      }
+
+   jumpReg = cg->evaluate(firstChild);
+   TR::addDependency(conditions, jumpReg, TR::RealRegister::x0, TR_GPR, cg);
```

Comment:
`x0` is the first argument register for private linkage.  You need to choose a volatile reg, `x8` through `x15`.